src/gsy_e/models/strategy/predefined_load.py

```diff
@@ -68,6 +135,9 @@ def __init__(self, daily_load_profile=None,
             update_interval = \
                 duration(minutes=ConstSettings.GeneralSettings.DEFAULT_UPDATE_INTERVAL)
 
+        if not hasattr(self, "_energy_params"):
+            self._energy_params = DefinedLoadEnergyParameters(
+                daily_load_profile, daily_load_profile_uuid)
```

Comment:
Nit, AFAIS this if condition is here (and in the `LoadHoursStrategy` init) because you want to configure different ways of initializing the _energy_params member.
A possible alternative would be to implement a private method `_setup_energy_params` in the base class which is called from the initializer always, and can be overridden in the child classes to define different implementation.

Reply:
As you might have noticed by the `_create_energy_parameters` I tried this approach and I unfortunately did not like it that much. Main reason was that each energy parameters class requires different arguments to be constructed, therefore I had to rely to *args / **kwargs for this method, which complicated the implementation.
If you really think that it will be better this way let me know and I will give it a try. Then we can talk again about it and keep it or revert it. I am suggesting this because it is a rather important change here so it probably is worth it to see both implementation options.

Reply:
Yeah unfortunately it would be a bit complicated to homogenize the interface of such method, in this case feel free to leave it as it is. Thanks!